tools/prepare_prebuilt_workers/prepare_prebuilt_workers.go

```diff
@@ -0,0 +1,134 @@
+//
+// This tool uses Go's text/template package for templating, see
+// https://pkg.go.dev/text/template for a description of the syntax.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+)
+
+// Tests contains the values for fields that are accessible by
+// flags.
+type Tests struct {
+	preBuiltImagePrefix string
+	testTag             string
+	dockerfileRoot      string
+	languagesToGitref   map[string]string
+}
+
+type langFlags []string
+
+func (l *langFlags) String() string {
+	var result string
+	for _, lang := range *l {
+		result = result + " " + lang
+	}
+	return result
+}
+
+func (l *langFlags) Set(value string) error {
+	*l = append(*l, value)
+	return nil
+}
+
+var languagesSelected langFlags
+
+func main() {
+	var test Tests
+
+	// specify the the image registry to push images
+	flag.StringVar(&test.preBuiltImagePrefix, "p", "", "image registry to push the image")
+
+	// specify the tag of prebuilt image
+	flag.StringVar(&test.testTag, "t", "", "tags for pre-built images, this unique tag to identify the images build and pushed in current test")
+
+	// specify the root path of Dockerfiles for prebuilt images
+	flag.StringVar(&test.dockerfileRoot, "r", "", "root directory of dockerfiles to build prebuilt images")
+
+	// specify the languages wish to run tests with
+	flag.Var(&languagesSelected, "l", "languages and its GITREF wish to run tests, example: cxx:master")
+
+	flag.Parse()
+
+	if test.preBuiltImagePrefix == "" {
+		log.Fatalf("failed to prepare prebuilt images: no registry provided, please provide a containr registry to store the images")
+	}
+
+	if test.testTag == "" {
+		log.Fatalf("failed to prepare prebuilt images: no image tag provided")
+	} else if len(test.testTag) > 128 {
+		log.Fatalf("failed to prepare prebuilt images: invalid tag name, a tag name may not start with a period or a dash and may contain a maximum of 128 characters.")
+	}
+
+	if test.dockerfileRoot == "" {
+		log.Fatalf("fail to prepare prebuilt images: no root directory for Dockerfiles provided")
+	}
+
+	if len(languagesSelected) == 0 {
+		log.Fatalf("failed to prepare prebuilt images: no language and its gitref pair specified, please provide languages and the GITREF as cxx:master")
+	}
+
+	test.languagesToGitref = map[string]string{}
+	for _, pair := range languagesSelected {
+		split := strings.Split(pair, ":")
+		if len(split) != 2 {
+			log.Fatalf("input error, please follow the format as language:gitref, for example: cxx:master")
+		}
+		test.languagesToGitref[split[0]] = split[1]
+	}
+
+	log.Println("selected language : GITREF")
+	log.Println(test.languagesToGitref)
+	//formattedMap, _ := json.MarshalIndent(test.languagesToGitref, "", "  ")
+	//log.Print(string(formattedMap))
+
+	for lang, gitRef := range test.languagesToGitref {
+		var image = fmt.Sprintf("%s/%s:%s", test.preBuiltImagePrefix, lang, test.testTag)
+		var dockerfileLocation = fmt.Sprintf("%s/%s/", test.dockerfileRoot, lang)
```

Comment:
For variable declarations that are also assignments, we should use the walrus operator `:=`

```suggestion
		image := fmt.Sprintf("%s/%s:%s", test.preBuiltImagePrefix, lang, test.testTag)
		dockerfileLocation := fmt.Sprintf("%s/%s/", test.dockerfileRoot, lang)
```